pytorch_lightning/trainer/trainer.py

```diff
@@ -174,6 +176,7 @@ def __init__(
         plugins: Optional[Union[PLUGIN_INPUT, List[PLUGIN_INPUT]]] = None,
         amp_backend: str = "native",
         amp_level: Optional[str] = None,
+        hmp_params: ["level", "verbose", "bf16_ops", "fp32_ops"] = None,
```

Comment:
how often will people change these? Is it sensible to have these as default and ask people to instantiate the plugin class themselves if they want to override?
We have this for other backends like FSDP or DeepSpeed already

Also: please no mutable defaults, this is one of python's quirks :)

Reply:
Using amp doesn't seem to be the right way. We might need to generalize this.